src/hooks/useProduct.ts

```diff
@@ -0,0 +1,47 @@
+import { useState, useEffect } from 'react';
+
+interface Product {
+	_id: string;
+	isActive: boolean;
+	price: string;
+	picture: string;
+	category: string;
+	name: string;
+}
+
+const useProduct = () => {
+	const [products, setProducts] = useState<Product[]>([]);
+	const [categories, setCategories] = useState<string[]>([]);
+
+	useEffect(() => {
+		fetch('./products.json')
+			.then((response) => {
+				return response.json();
+			})
+			.then((data) => {
+				const tempCategories: string[] = [];
+				for (let i = 0; i < data.length; i++) {
+					if (!tempCategories.includes(data[i].category)) {
+						tempCategories.push(data[i].category);
+					}
+				}
+				setCategories(tempCategories);
+
+				setProducts(data);
+			})
+			.catch((err: Error) => {
+				console.log(err.message);
+			});
+	}, []);
+
+	const product = {
```

Comment:
this not a correct variable name, this should be something like `productData`
in this case, use can simplify these codes like this:
```suggestion
	return {
           products,
	   setProducts,
           categories,
	   setCategories
	}
```

Reply:
Suggestion applied. Renamed product to productsData